frontend/lib/create_listing/view/create_listing.view.dart

```diff
@@ -0,0 +1,414 @@
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _submitForm() async {
+    setState(() {
+      _showValidationErrors = true;
+    });
+
+    final isFormValid = _formKey.currentState!.validate();
+    final hasImage = viewModel.hasImage;
+    final hasCondition = viewModel.condition.isNotEmpty;
+
+    if (isFormValid && hasImage && hasCondition) {
+      final success = await viewModel.createListing(
+        title: _titleController.text.trim(),
+        price: double.parse(_priceController.text),
+        description: _descriptionController.text.trim(),
+      );
+
+      if (success && mounted) {
+        context.pop();
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final appBar = AppBar(
+      leading: IconButton(
+        icon: const Icon(Icons.arrow_back),
+        onPressed: () => context.pop(),
+        color: const Color(0xFF2F9CCF),
+      ),
+      centerTitle: true,
+      title: appBarTitle,
+      actions: [
+        TextButton(
+          onPressed: _submitForm,
+          child: postButton,
+        ),
+      ],
+      backgroundColor: Colors.transparent,
+      elevation: 0,
+    );
+
+    final formContent = Form(
+      key: _formKey,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          _buildImageSelector(),
+          const SizedBox(height: 16),
+          _buildTextField('Title', _titleController),
+          const SizedBox(height: 16),
+          _buildTextField('Price', _priceController, isPrice: true),
+          const SizedBox(height: 16),
+          _buildConditionSelector(),
+          const SizedBox(height: 16),
+          _buildDescriptionField(),
+        ],
+      ),
+    );
+
+    final scrollableContent = SingleChildScrollView(
+      padding: const EdgeInsets.only(
+        left: 16,
+        right: 16,
+        top: 16,
+        bottom: 80,
+      ),
+      child: formContent,
+    );
+
+    final bottomButton = Align(
+      alignment: Alignment.bottomCenter,
+      child: Container(
+        color: Colors.white,
+        padding: const EdgeInsets.all(16),
+        child: ElevatedButton(
+          onPressed: _submitForm,
+          style: ElevatedButton.styleFrom(
+            backgroundColor: const Color(0xFF11384A),
+            minimumSize: const Size(double.infinity, 50),
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(25),
+            ),
+          ),
+          child: const Text(
+            'Post',
+            style: TextStyle(
+              color: Colors.white,
+              fontSize: 16,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    return Scaffold(
+      appBar: appBar,
+      body: Consumer<CreateListingViewModel>(
+        builder: (_, model, __) {
+          if (model.isLoading) {
+            return const Center(child: LoadingComponent());
```

Comment:
The loading component might come into play later when waiting for the API call to succeed but it shouldn't just appear every time we move to the new page.